templates/reservation/reservation.py

```diff
@@ -30,9 +32,22 @@ def __init__(self, name, guid=None, data=None):
     def validate(self):
         if not self.data.get('creationTimestamp'):
             self.data['creationTimestamp'] = time.time()
+        if not self.data.get('expiryTimestamp'):
+            self.data['expiryTimestamp'] = time.time()
```

Comment:
does that means all previous transaction will be marked as expired after we apply this patch ?

Reply:
well, in case of install, I am depending on install setting the extending it. However, I dont like the fact that we don't set the creationtimestamp in the data before creating the service to begin with. I'll add both to the data of the service before creation in the grid_broker. If I remove it those two lines from the validate, should I then raise an error if someone tries to extend a service where the expirytimestamp isnt set? this should theoritically not happen unless someone ceates the service themselves and not through grid_broker.

Reply:
Ok since I now set it in the data before creating a service, I changed the validate to raise an error if either aren't set. I think this makes more sense.